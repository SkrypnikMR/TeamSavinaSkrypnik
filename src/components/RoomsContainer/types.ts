type colors = {backgroundSecondary: string}

export type TStyledConatiner = {
    colors: colors;
    theme?: string;
    key: string;
}
