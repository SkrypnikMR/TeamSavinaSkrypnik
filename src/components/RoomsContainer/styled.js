import styled from 'styled-components';

export const StRoomsContainer = styled.div`
    background: ${({ colors, theme }) => colors[theme].backgroundSecondary};
    width: 30%;
    margin: 10px;
    max-height: 100%;
    display: flex;
    flex-direction: column;
    padding: 10px 0;
`;
