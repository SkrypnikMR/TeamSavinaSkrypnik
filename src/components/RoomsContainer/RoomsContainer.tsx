import React from 'react';
import {StRoomsContainer} from './styled';
import SingleRoom from '../SingleRoom'
import AddRoom from '../AddRoom';

const RoomsContainer = () =>{
    return(
        <StRoomsContainer >
            <SingleRoom />
            <SingleRoom />
            <SingleRoom />
            <AddRoom />
        </StRoomsContainer>
    )
}

export default RoomsContainer