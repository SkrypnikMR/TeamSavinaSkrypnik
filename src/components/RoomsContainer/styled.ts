import styled from 'styled-components';
import { white, turquous } from '../UI/baseLayout';

export const StRoomsContainer = styled.div`
    background: ${({ colors, theme }) => colors[theme].backgroundSecondary};
    width: 28%;
    margin: 10px;
    max-height: 100%;
    display: flex;
    flex-direction: column;
    padding: 10px 0;
    border-radius: 4px;
    overflow-y: scroll;
    overflow-x: hidden;
    box-shadow: 0 2px 4px rgba(0, 0, 0, .2);
    
  &::-webkit-scrollbar {
    border-radius: 10px;
    width: 10px;
    height: 40px;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${turquous};
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.2);
    background: ${turquous} radial-gradient(circle at 50% 0, rgba(${turquous},.65), rgba(${turquous},.35));
    box-shadow:
    inset rgba(0,0,0,.6) 0 -3px 8px,
    inset rgba(252,255,255,.7) 0 3px 8px,
    rgba(0,0,0,.8) 0 3px 8px -3px;
  }

  &::-webkit-scrollbar-track {
    margin: 3px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.2);
    border-radius: 10px;
    background-color: ${white};
  }
`;
