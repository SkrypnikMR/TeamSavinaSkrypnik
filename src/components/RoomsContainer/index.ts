import { connect } from 'react-redux';
import RoomsContainer from './RoomsContainer';
import { ApplicationState } from '../../store/types';
import { getRooms, getUserLogin } from '../../store/game/selectors';

const mapStateToProps = (state: ApplicationState) => ({
    rooms: getRooms(state),
    userLogin: getUserLogin(state),
});

export default connect(mapStateToProps)(RoomsContainer);
