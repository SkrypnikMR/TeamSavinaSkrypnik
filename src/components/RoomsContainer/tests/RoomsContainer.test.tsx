import React from 'react';
import RoomsContainer from '../RoomsContainer';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';

describe('RoomsContainer', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<RoomsContainer />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StRoomsContainer', () => {
        const component = mountSmart(<RoomsContainer/>);
        expect(component.find('styled__StRoomsContainer')).toHaveLength(1);
    });
    it('should render SingleRoom', () => {
        const component = mountSmart(<RoomsContainer/>);
        expect(component.find('SingleRoom')).toHaveLength(3);
    });
    it('should render AddRoom', () => {
        const component = mountSmart(<RoomsContainer/>);
        expect(component.find('AddRoom')).toHaveLength(1);
    });
});
