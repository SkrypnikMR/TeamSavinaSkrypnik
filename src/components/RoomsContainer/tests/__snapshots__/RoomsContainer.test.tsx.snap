// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomsContainer Should match snapshot 1`] = `"<div class=\\"styled__StRoomsContainer-sc-2jyga5-0 ilAtBM\\"><div class=\\"styled__StSingleRoom-sc-1b2p865-0 jzlaBL\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 jzlaBL\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 jzlaBL\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div></div>"`;
